build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -664,9 +664,12 @@ export async function setPackageDependencies(
 		if (dep !== undefined) {
 			const isSameReleaseGroup = MonoRepo.isSame(dep.pkg.monoRepo, pkg.monoRepo);
 			if (!isSameReleaseGroup || (updateWithinSameReleaseGroup && isSameReleaseGroup)) {
-				const dependencies = dev
+				let dependencies = dev
 					? pkg.packageJson.devDependencies
 					: pkg.packageJson.dependencies;
+				if (dependencies === undefined) {
+					dependencies = {};
```

Comment:
I don't think this works. If you do this, the line below which modifies dependencies modifies this new object, not the actual packageJson object.

It would be good to have a test for this showing that it works: that makes code review easier as we don't have to look for bugs which have test coverage.

Reply:
Note that testing this cleanly would likely require factoring out the updating logic like the typeTests code does.